### PR TITLE
Add waterproof to ci

### PR DESCRIPTION
**Pull request by jim-portegies**
We would like to have our plugin "Waterproof" https://github.com/impermeable/coq-waterproof added to the CI for Coq. We have tried to follow the instructions from https://github.com/coq/coq/blob/master/dev/ci/README-users.md . It can be that we made some mistakes, but would like to take you up on the offer on that page for help finishing the pull request.

### CODE REVIEW

File: dev/ci/ci-waterproof.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+set -e
+
+ci_dir="$(dirname "$0")"
+. "{ci_dir}/ci-common.sh"
+
+git_download waterproof
+
+if [ "$DOWNLOAD_ONLY" ]; then exit 0; fi
+
+( cd "${CI_BUILD_DIR}/coq-waterproof"
```
Comment (SkySkimmer) [inline dev/ci/ci-waterproof.sh:12]:
```suggestion
( cd "${CI_BUILD_DIR}/waterproof" 
```
the directory name comes from the name we give to `git_download` not the name in the url

**Review by SkySkimmer — COMMENTED** [review]
With my changes, running multiple times (as is common when making overlays) produces
~~~
+ dune build -p coq-waterproof
Warning: Coq Language Versions lower than 0.8 have been deprecated in Dune
3.8 and will be removed in an upcoming Dune version.
Error: Multiple rules generated for _build/default/src/waterproof.a:
- file present in source tree
- src/dune:1
Hint: rm -f src/waterproof.a
Error: Multiple rules generated for _build/default/theories/Automation.glob:
- file present in source tree
- theories/dune:1
Hint: rm -f theories/Automation.glob
~~~
and some more "multiple rules" for globs

Is there a reason to build partially with dune and partially with coq_makefile?
Probably you should pick one.

**Comment by SkySkimmer** [comment]
> Probably you should pick one.

(I mean for the CI script)

**Comment by jim-portegies** [comment]
> > Probably you should pick one.
> 
> (I mean for the CI script)

There is a reason in the sense that this order of building (and using both make and dune) allowed us to get our library to work with coq-lsp.

**Comment by SkySkimmer** [comment]
I'd prefer reruns to avoid rebuilding when nothing changed.

**Comment by SkySkimmer** [comment]
>There is a reason in the sense that this order of building (and using both make and dune) allowed us to get our library to work with coq-lsp.

cc @ejgallego for dune / lsp help

**Comment by jim-portegies** [comment]
I have removed dune build and dune install for now.

I tried to build everything with dune, but I'm running into an error `path outside the workspace` which seems to be caused by a line `Declare ML Module "coq-core.plugins.ltac2".` in one of our `.v` files. This error seems to have come in with version 8.17.1. I have a tiny example on the branch "tiny" of the coq-waterproof repository: https://github.com/impermeable/coq-waterproof/tree/tiny .

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Comment by jim-portegies** [comment]
@SkySkimmer  I changed somethings in the coq-waterproof and it seems to have fixed the dune issue. Would you prefer that I make a new pull request for build with dune or is it better to incorporate it here?

**Comment by SkySkimmer** [comment]
I'm not sure what you mean. Do what you want and we will see how it goes.

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Comment by jim-portegies** [comment]
I see the run failed, I am not 100% sure what is the problem, but I've now used dune lang 3.6 on the coq-waterproof master branch in the hope to fix the error...

**Comment by Zimmi48** [comment]
That's because you are using Dune 3.9 and our edge switch only has Dune 3.6. You can bump the Dune version in the edge switch by modifying the [Dockerfile](https://github.com/coq/coq/blob/master/dev/ci/docker/bionic_coq/Dockerfile). Make sure to regenerate the image key in the gitlab-ci.yml file as documented.

**Comment by SkySkimmer** [comment]
job retry https://gitlab.com/coq/coq/-/jobs/4655915005

File: dev/ci/ci-basic-overlay.sh [hunk 2]
```diff
@@ -394,3 +394,8 @@ project tactician "https://github.com/coq-tactician/coq-tactician" "coqdev"
 # Ltac2 compiler
 ########################################################################
 project ltac2_compiler "https://github.com/SkySkimmer/coq-ltac2-compiler" "main"
+
+########################################################################
+# Waterproof
+########################################################################
+project waterproof "https://github.com/impermeable/coq-waterproof" "master"
```
[the remark this answers is not in the document]
Comment (jim-portegies) [reply dev/ci/ci-basic-overlay.sh]: Agreed, I changed the name to `coq-master`

**Comment by jim-portegies** [comment]
@SkySkimmer shall we try to rerun the CI?

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Comment by jim-portegies** [comment]
@SkySkimmer I have removed a trailing whitespace in the hope to fix the lint error. Do we need to rerun the CI?

**Comment by jim-portegies** [comment]
@SkySkimmer what would be the further steps for the pull request to be accepted?

**Review by SkySkimmer — APPROVED** [review]

**Comment by SkySkimmer** [comment]
>@SkySkimmer what would be the further steps for the pull request to be accepted?

Me coming back from holidays today ;)
Also please squash

**Comment by jim-portegies** [comment]
@SkySkimmer sorry how do I squash?

**Comment by SkySkimmer** [comment]
https://stackoverflow.com/questions/5189560/how-do-i-squash-my-last-n-commits-together/5189600#5189600

**Comment by SkySkimmer** [comment]
@coqbot merge now

**Pull request merged**